addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -165,6 +166,38 @@ private void checkMetaElementsMatches(Source source) {
         }
     }
 
+    private void checkDomElementMatches(Source source) {
+        Document doc = Jsoup.parse(String.valueOf(source));
+
+        for (Map<String, Map<String, Map<String, AppPattern>>> map : currentApp.getDom()) {
+            for (Map.Entry<String, Map<String, Map<String, AppPattern>>> ap : map.entrySet()) {
+                for (Map.Entry<String, Map<String, AppPattern>> p : ap.getValue().entrySet()) {
+                    for (Map.Entry<String, AppPattern> pa : p.getValue().entrySet()) {
+                        Elements links = doc.select(ap.getKey());
+                        for (org.jsoup.nodes.Element link : links) {
```

Comment:
Could you make the variable names a bit more meaningful here? Ex: 
- `map` ➡️ `domSelectorMap`
- `link` & `links` ➡️ `selectedElement` & `selectedElements`
- etc